src/librustc_mir/callgraph.rs

```diff
@@ -0,0 +1,242 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+//! MIR-based callgraph.
+//!
+//! This only considers direct calls
```

Comment:
This is not an issue, but it's interesting that this is an "optimistic" call graph -- i.e., it is only those things that we **know** will be called. Often people would assume a pessimistic approach. Maybe worth highlighting just a bit in the comment here.